sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/InvalidCastToInterface.cs

```diff
@@ -175,3 +175,92 @@ public void Method<T>(T generic) where T : new()
         }
     }
 }
+
+namespace Tests.Diagnostics
+{
+    namespace CSharp8
+    {
+        public class NullCoalescenceAssignment
+        {
+            public void NullCoalescenceAssignment_Null()
+            {
+                int? i1 = null;
+                i1 ??= (int)i1; // Noncompliant {{Nullable is known to be empty, this cast throws an exception.}}
+            }
+
+            public void NullCoalescenceAssignment_NotNull()
+            {
+                int? i1 = 1;
+                i1 ??= (int)i1;
+            }
+        }
+
+        public class SwitchExpressions
+        {
```

Comment:
I suggest to add pair of tests like this to illustrate FN behaviour as documented in new issue here #2949 
```
            public int CorrectCheck(int? val)
            {
                return val.HasValue switch
                {
                    true => (int)val,
                    _ => 0
                };
            }

            public int WrongCheck_Noncompliant(int? val)
            {
                return val.HasValue switch
                {
                    false => (int)val,  // FN
                    _ => 0
                };
            }
```